GoogleUtilities/Environment/NetworkInfo/GULNetworkInfo.m

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GoogleUtilities/Environment/Public/GoogleUtilities/GULNetworkInfo.h"
+
+#import <Foundation/Foundation.h>
+
+#import <SystemConfiguration/SystemConfiguration.h>
+
+#import <TargetConditionals.h>
+#if __has_include("CoreTelephony/CTTelephonyNetworkInfo.h") && !TARGET_OS_MACCATALYST && \
+                  !TARGET_OS_OSX && !TARGET_OS_TV
+#define TARGET_HAS_MOBILE_CONNECTIVITY
+#import <CoreTelephony/CTCarrier.h>
+#import <CoreTelephony/CTTelephonyNetworkInfo.h>
+#endif
+
+@implementation GULNetworkInfo
+
+#ifdef TARGET_HAS_MOBILE_CONNECTIVITY
++ (CTTelephonyNetworkInfo *)getNetworkInfo {
+  static CTTelephonyNetworkInfo *networkInfo;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    networkInfo = [[CTTelephonyNetworkInfo alloc] init];
+  });
+  return networkInfo;
+}
+#endif
+
++ (NSString *_Nullable)getNetworkMobileCountryCode {
+#ifdef TARGET_HAS_MOBILE_CONNECTIVITY
+  CTTelephonyNetworkInfo *networkInfo = [GULNetworkInfo getNetworkInfo];
+  CTCarrier *provider = networkInfo.subscriberCellularProvider;
+  return provider.mobileCountryCode;
+#endif
+  return nil;
+}
+
++ (NSString *_Nullable)getNetworkMobileNetworkCode {
+#ifdef TARGET_HAS_MOBILE_CONNECTIVITY
+  CTTelephonyNetworkInfo *networkInfo = [GULNetworkInfo getNetworkInfo];
+  CTCarrier *provider = networkInfo.subscriberCellularProvider;
+  return provider.mobileNetworkCode;
+#endif
+  return nil;
+}
+
++ (NSString *_Nullable)formatMcc:(NSString *)mcc andMNC:(NSString *)mnc {
```

Comment:
Please document mcc and mnc here or use longer variable names.

Reply:
Awesome - added the blurb from the header